src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
+    public MyLinkedList() {
+        firstNode = null;
+        lastNode = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            firstNode = newNode;
+            lastNode = newNode;
+        } else if (index == 0) {
+            newNode.next = firstNode;
+            firstNode.prev = newNode;
+            firstNode = newNode;
+        } else if (index == size) {
+            newNode.prev = lastNode;
+            lastNode.next = newNode;
+            lastNode = newNode;
+        } else {
+            Node<T> current = firstNode;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            newNode.prev = current.prev;
+            newNode.next = current;
+            current.prev.next = newNode;
+            current.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> current = firstNode;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
```

Comment:
create a private method to find a Node,
this logic is repeated in the code
```suggestion
        Node<T> current = getCurrentNodeBy(index)
```